src/he/conn.c

```diff
@@ -146,7 +146,7 @@ he_conn_t *he_conn_create(void) {
 void he_conn_destroy(he_conn_t *conn) {
   if(conn) {
     // Free up all cached fragments
-    for(size_t i = 0; i < sizeof(conn->frag_table.entries) / sizeof(he_fragment_entry_t); i++) {
+    for(size_t i = 0; i < (sizeof(conn->frag_table.entries) / sizeof(he_fragment_entry_t *)); i++) {
```

Comment:
Sorry for slow reply, just catching up after vacation.
I think we could have avoided this bug in the first place, and prevent accidentally regressing in the future if we wrote the code like this instead:

```C
size_t num_entries = sizeof(conn->frag_table.entries) / sizeof(he_fragment_entry_t *);
for(size_t i = 0; i < num_entries; i++) {
```
I'm sure the compiler will inline this, so no negative performance impact :).  However it's a positive readability impact! 

Shall I raise a PR?

Reply:
The compiler will indeed inline this - and it'll just be a static value (computed at compile time) :slightly_smiling_face:

Reply:
Hi Andre, I'm not sure how writing it that way would have avoided the issue. But sure it does improve readability so go for it.

The compiler will almost certainly inline this a compile time constant.

Reply:
> `size_t num_entries = sizeof(conn->frag_table.entries) / sizeof(he_fragment_entry_t *);`

I suggest:
```
size_t num_entries = sizeof(conn->frag_table.entries) / sizeof(conn->frag_table.entries[0]);
```
which avoids an issue if you get the type of the array member wrong.

Reply:
This is also the sort of thing where you commonly see:
```c
#define ARRAY_LEN(x) (sizeof(x) / sizeof(x[0]))
```
used throughout a code base...